src/geoaxis.jl

```diff
@@ -0,0 +1,116 @@
+
+In the call signature, `args...` is a standard figure location, e.g., `fig[1,1]` as given in
+`Axis`. The keyword arguments decide the geospatial projection:
+
+* `source = "+proj=longlat +datum=WGS84", dest = "+proj=eqearth"`: These two keywords
+  configure the map projection to be used for the given field using Proj4.jl.
+  See also online the section [Changing central longitude](@ref) for data that may not 
+  span the (expected by default) longitude range from -180 to 180.
+* `transformation = Proj4.Transformation(source, dest, always_xy=true)`: Instead of
+  `source, dest` you can directly use the Proj4.jl package to define the projection.
+* `coastlines = true`: Whether to plot coastlines. Coastlines are plotted using
+  ```
+  lines!(ax, GeoMakie.coastlines(); coastkwargs...)
+  ```
+* `coastkwargs = NamedTuple()` Keywords propagated to the coastline plot (which is a line plot).
+* `lonticks = -180:60:180, latticks = -90:30:90` ticks for the longitude and latitude
+  dimensions. The grid lines of the axis are also spanning these tick values.
+* `hidespines = true` Hide the axis spines (rectangle surrounding the axis).
+
+## Example
+```julia
+using GeoMakie, GLMakie
+
+lons = -180:180
+lats = -90:90
+field = [exp(cosd(l)) + 3(y/90) for l in lons, y in lats]
+
+# Surface example
+fig = Figure()
+ax = GeoAxis(fig[1,1])
+el = surface!(ax, lons, lats, field)
+display(fig)
+
+# Scatter example
+slons = rand(lons, 2000)
+slats = rand(lats, 2000)
+sfield = [exp(cosd(l)) + 3(y/90) for (l,y) in zip(slons, slats)]
+
+fig = Figure()
+ax = GeoAxis(fig[1,1])
+el = scatter!(slons, slats; color = sfield)
+display(fig)
+```
+"""
+function GeoAxis(args...; 
+        source = "+proj=longlat +datum=WGS84", dest = "+proj=eqearth",
+        transformation = Proj4.Transformation(source, dest, always_xy=true),
+        coastlines = true, coastkwargs = NamedTuple(),
+        lonticks = -180:60:180,
+        latticks = -90:30:90,  
+        hidespines = true,
+        kw...
+    )
+
+    # Generate Axis instance
+    ax = Axis(args...; aspect = DataAspect(), interpolate_grid = true, kw...)
+    
+    # TODO: I don't know what the following comment means, but it was here
+    # before @Datseris did the `GeoAxis` PR.
+
+    # ax = Axis(args...; interpolate_grid=true)
+    # interpolate_grid would need to be implemented in the axis code, but that should be fairly straightforward 
+    # needed to make the grid warp correctly
+
+    # Set axis transformation
+    ptrans = Makie.PointTrans{2}(transformation)
+    ax.scene.transformation.transform_func[] = ptrans
+
+    # Set axis limits
+    # This seems a bit inefficient, but it is the only way to ensure 
+    # correct limits independently of the projection.
+    lons = -180:180
+    lats = -90:90
+    points = [Point2f0(lon, lat) for lon in lons, lat in lats]
+    rectLimits = FRect2D(Makie.apply_transform(ptrans, points))
+    limits!(ax, rectLimits)
+
+    # Plot coastlines
+    if coastlines
```

Comment:
no sure if coastlines should be here by default. They are already one-liner if needed afterwards. Plus, here as it is it will be hard to modified afterwards, like color, linestyle, width. Probably, it will be better to do an example showing how to add them later in the work-flow.

Reply:
You know what, actually you are right. At the moment there is the extra keyword for coastline keyword arguments, but maybe better to split it away. I will do this in next commit.